kratos/factories/standard_preconditioner_factory.cpp

```diff
@@ -43,11 +43,24 @@ namespace Kratos
         static auto ILU0PreconditionerFactory= StandardPreconditionerFactory<SpaceType,LocalSpaceType,ILU0PreconditionerType>();
         static auto ILUPreconditionerFactory= StandardPreconditionerFactory<SpaceType,LocalSpaceType,ILUPreconditionerType>();
 
+        // Setting default register name
+        KRATOS_SET_REGISTER_SOURCE("KratosMultiphysics");
```

Comment:
why is this needed? The kernel does this already

Reply:
The new macro I introduced assumes that you are going to register things inside `KratosApplication`class (which is 99% of the case), hence `mApplicationName` exists.

For preconditioners and linear solvers in the core, and possibly others, registering is done outside the application class itself, and I need a way to track down where it belongs.

I will mark this as solved as is basically an extension of the comment below so we keep it centralized